tests/apache/livy/hooks/test_livy.py

```diff
@@ -14,279 +13,233 @@
-    response = await hook._do_api_call_async(endpoint=GET_RUN_ENDPOINT, headers={})
-    assert response == {"Response": "Unexpected HTTP Method: abc", "status": "error"}
-
-
-@pytest.mark.asyncio
-@mock.patch("astronomer.providers.apache.livy.hooks.livy.aiohttp.ClientSession")
-@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_connection")
-async def test_do_api_call_async_with_type_error(mock_get_connection, mock_session):
-    """
-    Asserts the _do_api_call_async for TypeError.
-    """
-
-    async def mock_fun(arg1, arg2, arg3, arg4):
-        return {"random value"}
-
-    mock_session.return_value.__aexit__.return_value = mock_fun
-    mock_session.return_value.__aenter__.return_value.patch.return_value.json.return_value = {}
-    hook = LivyHookAsync(livy_conn_id="livy_default")
-    hook.method = "PATCH"
-    hook.retry_limit = 1
-    hook.retry_delay = 1
-    hook.http_conn_id = mock_get_connection
-    with pytest.raises(TypeError):
-        await hook._do_api_call_async(endpoint="", data="test", headers=mock_fun, extra_options=mock_fun)
-
-
-@pytest.mark.asyncio
-@mock.patch("astronomer.providers.apache.livy.hooks.livy.aiohttp.ClientSession")
-@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_connection")
-async def test_do_api_call_async_with_client_response_error(mock_get_connection, mock_session):
-    """
-    Asserts the _do_api_call_async for Client Response Error.
-    """
-
-    async def mock_fun(arg1, arg2, arg3, arg4):
-        return {"random value"}
-
-    mock_session.return_value.__aexit__.return_value = mock_fun
-    mock_session.return_value.__aenter__.return_value.patch.return_value.json.side_effect = (
-        ClientResponseError(
-            request_info=RequestInfo(url="example.com", method="PATCH", headers=multidict.CIMultiDict()),
-            status=500,
-            history=[],
+class TestLivyHookAsync:
+    @pytest.mark.asyncio
+    @mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.run_method")
+    async def test_get_batch_state_running(self, mock_run_method):
+        """Asserts the batch state as running with success response."""
+        mock_run_method.return_value = {"status": "success", "response": {"state": BatchState.RUNNING}}
+        hook = LivyHookAsync(livy_conn_id="livy_default")
+        state = await hook.get_batch_state(BATCH_ID)
+        assert state == {
+            "batch_state": BatchState.RUNNING,
+            "response": "successfully fetched the batch state.",
+            "status": "success",
+        }
+
+    @pytest.mark.asyncio
+    @mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.run_method")
+    async def test_get_batch_state_error(self, mock_run_method):
+        """Asserts the batch state as error with error response."""
+        mock_run_method.return_value = {"status": "error", "response": {"state": "error"}}
+        hook = LivyHookAsync(livy_conn_id="livy_default")
```

Comment:
All these values can be variables. since it's repeated in multiple places. `livy_conn_id="livy_default"`

Reply:
we could have a variable for 'livy_default'